packages/api/repository/ApplicationRepository.ts

```diff
@@ -39,8 +39,9 @@ export class ApplicationRepository extends Repository<Application> {
 
   // DELETE application (hard delete)
   async deleteApplication(applicationId: string): Promise<Application> {
-    console.log("Deleting: " + applicationId);
+    // Check if the app exists, if it doesn't findOneOrFail will throw 404 to error handling middleware
     const appToDelete = await this.findOneOrFail({ id: applicationId });
+    console.log("Deleting: " + applicationId);
     const deleteResult = await this.delete(applicationId);
     console.log(deleteResult);
     return appToDelete;
```

Comment:
This could probably be updated to return `void` since no data is to be returned. This relates to a comment in  PR #549 ([comment link](https://github.com/dod-ccpo/atat-web-api/pull/549/files#r772585887)).
```suggestion
  async deleteApplication(applicationId: string): Promise<void> {
    // Check if the app exists, if it doesn't findOneOrFail will throw 404 to error handling middleware
    const appToDelete = await this.findOneOrFail({ id: applicationId });
    console.log("Deleting: " + applicationId);
    const deleteResult = await this.delete(applicationId);
    console.log(deleteResult);
    console.log("Deleted application: " + appToDelete);
```

Reply:
In theory I agree, since API spec specifies to return 204 NO_CONTENT for operation `deleteApplication`.
However, confusingly, it also says the content should be schema object `Application`.

It seems this is true for all `DELETE` operations with the exception of the operation for deleting a task order file.  They all return their respective schema objects, so this appears to be intentional.

I think we do need to return the deleted Application unless we change the API spec.

>       responses:
>        '204':
>          description: Successful operation
>          content:
>            application/json:
>              schema:
>                $ref: '#/components/schemas/Application'